enforcer-rules/src/main/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRule.java

```diff
@@ -123,34 +128,32 @@ public void execute(@Nonnull EnforcerRuleHelper helper) throws EnforcerRuleExcep
 
       try {
         LinkageChecker linkageChecker = LinkageChecker.create(classpath, directDependencies);
-        LinkageCheckReport linkageReport = linkageChecker.findLinkageErrors();
-        int errorCount =
-            linkageReport.getJarLinkageReports().stream()
-                .mapToInt(JarLinkageReport::getErrorCount)
-                .sum();
+        ImmutableSetMultimap<ClassFile, SymbolProblem> symbolProblems = linkageChecker
+            .findSymbolProblems();
         if (reportOnlyReachable) {
-          ImmutableList<JarLinkageReport> reachableErrorReports =
-              linkageReport.getJarLinkageReports().stream()
-                  .map(JarLinkageReport::reachableErrors)
-                  .collect(toImmutableList());
-          linkageReport = LinkageCheckReport.create(reachableErrorReports);
-          int reachableErrorCount =
-              reachableErrorReports.stream()
-                  .mapToInt(JarLinkageReport::getErrorCount)
-                  .sum();
-          errorCount = reachableErrorCount;
+          ClassReferenceGraph classReferenceGraph = linkageChecker.getClassReferenceGraph();
+          symbolProblems = symbolProblems.entries().stream().filter(
```

Comment:
This is hard to follow. Can it be broken up into multiple smaller statements, possibly without reusing a variable?

Reply:
The indentation was incorrect. Does this look good now?